File: Ais.IO/StructT.cpp
#include "pch.h"
#include "StructT.h"

void ModifyStruct(MyStruct* s) {
    if (s) {
        s->id += 100;
        s->value *= 2.0f;
    }
}

void EnumWindowsMock(EnumCallback callback) {
    if (!callback) return;

    // 模擬一些窗口數據
    WindowInfo windows[] = {
        {1, "Window 1"},
        {2, "Window 2"},
        {3, "Window 3"}
    };

    // 遍歷窗口並調用回調函數
    for (const auto& win : windows) {
        callback(&win);
    }
}

File: Ais.IO.Csharp.Command/StructT.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;
using System.Threading.Tasks;
using Ais.IO.Csharp;

namespace Ais.IO.Csharp.Command
{
    internal class StructT
    {
        public static void GetStruct()
        {
            StructTInterop.MyStruct myStruct = new StructTInterop.MyStruct { id = 1, value = 1.5f };
            Console.WriteLine($"Before: id={myStruct.id}, value={myStruct.value}");

            StructTInterop.ModifyStruct(ref myStruct);

            Console.WriteLine($"After: id={myStruct.id}, value={myStruct.value}");

            StructTInterop.Callback = (ref StructTInterop.WindowInfo window) => Console.WriteLine($"Window ID: {window.id}, Title: {window.title}");
            StructTInterop.EnumWindowsMock(StructTInterop.Callback);
        }
    }
}
